atomic_reactor/plugins/pre_cancel_builds_on_autorebuild.py

```diff
@@ -0,0 +1,84 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+from osbs.api import OSBS
+from osbs.conf import Configuration
+from osbs.exceptions import OsbsResponseException
+from atomic_reactor.util import get_build_json
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.plugins.pre_check_and_set_rebuild import is_rebuild
+
+
+class CancelBuildsOnAutoRebuild(PreBuildPlugin):
+    """
+    If the current build is rebuild, we need to query OSBS to make sure there
+    are no other builds currently running for this container and build target
+    """
+    key = 'cancel_builds_on_autorebuild'
+
+    # FIXME - Unsure if this should be allowed to fail
```

Comment:
I think it should not be allowed to fail. If this plugins fails to cancel an already running build, it might as well cancel itself. Having two builds pushing content to the same repo simultaneously can cause further problems.